File: tx/java/pando-tx/src/main/java/org/pando/tx/types/TxOutput.java
package org.pando.tx.types;

import java.math.BigInteger;
import org.ethereum.util.RLP;

public final class TxOutput {

    public byte[] address;
    public Coins  coins;

    public TxOutput(byte[] address, BigInteger pandoWei, BigInteger ptxWei) {
        this.address = address;
        this.coins = new Coins(pandoWei, ptxWei);
    }

    public byte[] rlpEncode() {
        byte[] rlpEncoded = RLP.encodeList(
            RLP.encode(this.address),
            this.coins.rlpEncode());
        return rlpEncoded;
    }
}

File: tx/javascript/pando-tx/src/tx/types/txoutput.js
import {Coins} from './coins';

export class TxOutput {
    constructor(address, pandoWei, ptxWei) {
        this.address = address;
        this.coins = new Coins(pandoWei, ptxWei);
    }

    rlpInput(){
        let rplInput = [
            this.address.toLowerCase(),
            this.coins.rlpInput()
        ];

        return rplInput;
    }
}